src/driver.ts

```diff
@@ -221,34 +224,49 @@ export class AppiumFlutterDriver extends BaseDriver<FlutterDriverConstraints> {
       ...JSON.parse(JSON.stringify(args)),
     );
 
-    // HACK for eliminatin socket hang up by waiting 1 sec
     await sleep(1000);
-    // @ts-ignore
-    console.log('PageSource', await this.proxydriver.getPageSource());
+    const { appPackage, bundleId } = this.proxydriver.opts as any;
+    const packageName = appPackage || bundleId;
+    const callbacks =
+      this.proxydriver instanceof AndroidUiautomator2Driver
+        ? {
+            portForwardCallback: androidPortForward,
+            portReleaseCallback: androidRemovePortForward,
+          }
+        : this.proxydriver.isRealDevice()
+          ? {
+              portForwardCallback: iosPortForward,
+              portReleaseCallback: iosRemovePortForward,
+            }
+          : {};
+    const systemPort =
+      this.proxydriver instanceof XCUITestDriver &&
+      !this.proxydriver.isRealDevice()
+        ? null
+        : await getFreePort();
```

Comment:
system port should be configurable from capabilities. there are examples of client machines where firewalls only allow to open only particular port ranges

Reply:
I have created a backlog to deal with customising options with capabilities. https://github.com/AppiumTestDistribution/appium-flutter-integration-driver/issues/7

Will tackle this as part for it as it need additional changes.